src/ziggurat/messaging/consumer.clj

```diff
@@ -1,20 +1,20 @@
 (ns ziggurat.messaging.consumer
-  (:require [ziggurat.mapper :as mpr]
-            [ziggurat.message-payload :as mp]
-            [clojure.tools.logging :as log]
+  (:require [clojure.tools.logging :as log]
             [langohr.basic :as lb]
             [langohr.channel :as lch]
             [langohr.consumers :as lcons]
-            [ziggurat.kafka-consumer.consumer-handler :as ch]
             [schema.core :as s]
-            [sentry-clj.async :as sentry]
             [taoensso.nippy :as nippy]
             [ziggurat.config :refer [get-in-config]]
+            [ziggurat.kafka-consumer.consumer-handler :as ch]
+            [ziggurat.mapper :as mpr]
+            [ziggurat.message-payload :as mp]
             [ziggurat.messaging.connection :refer [connection]]
-            [ziggurat.sentry :refer [sentry-reporter]]
-            [ziggurat.messaging.util :refer :all]
+            [ziggurat.messaging.util :refer [prefixed-queue-name prefixed-channel-name]]
             [ziggurat.metrics :as metrics]
-            [ziggurat.util.error :refer [report-error]]))
+            [ziggurat.middleware.default :as mw]
+            [ziggurat.util.error :refer [report-error]])
+  (:import [com.ziggurat.proto MessagePayloadProto$MessagePayload]))
 
 (defn- convert-to-message-payload
```

Comment:
We can remove this method. This is adding unnecessary complexity to the deserialization logic.

**The downside** : This code will no longer be backward compatible with 2.x. But, in favor of simplicity of the code (and the fact that we haven't broken B.C. till 3.14.0) allows us some freedom to remove this method.

Reply:
done